### PR TITLE
Add pane rename command

### PR DESCRIPTION
**Pull request by cweill**
## Motivation
Pane names are currently fixed after creation, which makes it awkward to relabel long-lived panes as their role changes. This adds a dedicated rename command while keeping pane references unambiguous by rejecting duplicate names across the session.

## Summary
- add `amux rename <pane> <new-name>` to the CLI/session command tables and server command registry
- update the server-side layout command layer to rename a resolved pane, reject duplicate session-wide names, and emit a layout broadcast
- prune stale pane-name event subscriptions when a pane is renamed so old `events --pane <name>` filters do not accidentally match a future pane that reuses the name
- add CLI parser/help coverage plus queued server command tests for successful rename and duplicate-name rejection across windows

## Testing
- `go test ./internal/server/commands/layout -run 'TestRename' -count=100`
- `go test ./internal/server -run 'TestQueuedCommandRenamePane' -count=100`
- `go test ./internal/cli -run 'TestResolveCanonicalSessionCommand|TestMaybePrintCommandHelp|TestRunMainDispatchesCommands' -count=100`
- `go test ./... -timeout 120s`

## Review focus
- the duplicate-name check is session-wide rather than window-local because pane refs must stay unique across the session
- rename now drops stale pane-name event subscriptions keyed to the old name; look at that behavior if you want to verify the cleanup boundary

Closes LAB-1125

### CODE REVIEW

File: internal/server/commands_layout.go [hunk 1]
```diff
@@ -541,6 +545,32 @@ func cmdFocus(ctx *CommandContext) {
...
+	return toCommandResult(ctx.Sess.enqueueCommandMutation(func(mctx *MutationContext) commandMutationResult {
+		pane, _, err := mctx.resolvePaneAcrossWindowsForActor(actorPaneID, paneRef)
+		if err != nil {
+			return commandMutationResult{err: err}
+		}
+		for _, candidate := range mctx.Panes {
+			if candidate.Meta.Name == name {
+				return commandMutationResult{err: fmt.Errorf("pane %q already exists", name)}
+			}
+		}
```
[the remark this answers is not in the document]
Comment (cweill) [reply internal/server/commands_layout.go:562]: Fixed in 65d5d58. Self-rename is now treated as a no-op, the duplicate scan skips the pane being renamed, and TestQueuedCommandRenamePane now covers rename-to-self without a layout generation bump.

**Comment by cweill** [comment]
Addressed the bot feedback from issue comments https://github.com/weill-labs/amux/pull/697#issuecomment-4234443632 and https://github.com/weill-labs/amux/pull/697#issuecomment-4234455033 in 65d5d58:

- self-rename no longer trips the duplicate-name guard
- `amux rename <pane> <same-name>` now succeeds as a true no-op instead of returning `already exists`
- added a regression case in `TestQueuedCommandRenamePane` to verify the command leaves the name unchanged and does not bump layout generation
